source/s3_auto_ranged_get.c

```diff
@@ -161,6 +188,28 @@ static bool s_s3_auto_ranged_get_update(
             goto has_work_remaining;
         }
 
+        if (auto_ranged_get->synced_data.get_without_range) {
+            if (out_request == NULL) {
```

Comment:
Move this conditional right after the next one, ie, if(auto_ranged_get->synced_data.get_without_range_sent).  At this point we don't know for certain there is actually work remaining.